File: atividades_verde/JAVA/Aquecimento_Iterativo.java
import java.util.Scanner;
public class Aquecimento_Iterativo {
    private static int numMaiusc(String x) {
        int contador = 0;
        for(int i = 0; i < x.length(); i++) {
            if(x.charAt(i) >= 'A' && x.charAt(i) <= 'Z') 
                contador++;
        }
        return contador;
    }
    private static boolean eIgual(String x, String y) {
        boolean resposta = false;
        if(x.length() == y.length()) {
            resposta = true;
            for(int i = 0; i < x.length(); i++) {
                if(x.charAt(i) != y.charAt(i)) {
                    resposta = false;
                    i = x.length();   
                }
            }
        }
        return resposta;
    }
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        String palavra = scan.nextLine();
        while (!eIgual(palavra, "FIM")) {
            int resposta = numMaiusc(palavra);
            System.out.println(resposta);
            palavra = scan.nextLine();
        }
    }
}

File: atividades_verde/C/aquecimento_iterativo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

int numMaiusc(char x[]) {
    int contador = 0;
    for(int i = 0; i < strlen(x); i++) {
        if(x[i] >= 'A' && x[i] <= 'Z') {
            contador++;
        }
    }
    return contador;
}
bool eIgual(char x[], char y[]) {
    bool resposta = false;
    if(strlen(x) == strlen(y)) {
        resposta = true;
        for(int i = 0; i < strlen(x); i++) {
            if(x[i] != y[i]) {
                resposta = false;
                i = strlen(x);
            }
        }
    }
    return resposta;
}
int main() {
    char palavra[500];
    scanf("%[^\n]", palavra);
    getchar();
    while(!eIgual(palavra, "FIM")) {
        int resposta = numMaiusc(palavra);
        printf("%d\n", resposta);
        scanf("%[^\n]", palavra);
        getchar();
    }
    return 0;
}